Tests/AK/TestString.cpp

```diff
@@ -330,25 +330,33 @@ TEST_CASE(equals_ignoring_case)
         String string1 {};
         String string2 {};
 
-        EXPECT(MUST(string1.equals_ignoring_case(string2)));
+        EXPECT(string1.equals_ignoring_case(string2));
     }
     {
         auto string1 = MUST("abcd"_string);
         auto string2 = MUST("ABCD"_string);
         auto string3 = MUST("AbCd"_string);
         auto string4 = MUST("dcba"_string);
+        auto string5 = MUST("abce"_string);
+        auto string6 = MUST("abc"_string);
 
-        EXPECT(MUST(string1.equals_ignoring_case(string2)));
-        EXPECT(MUST(string1.equals_ignoring_case(string3)));
-        EXPECT(!MUST(string1.equals_ignoring_case(string4)));
+        EXPECT(string1.equals_ignoring_case(string2));
```

Comment:
with the non-allocation version, can we wrap the tests in a `NoAllocationGuard`?

Reply:
I don't find that necessary, if this ever needs to allocate again, it'll return `ErrorOr` again and then must explicitly be handled.